covid19_sfbayarea/news/utils.py

```diff
@@ -114,6 +114,10 @@ def parse_datetime(date_string: str, timezone: Optional[tzinfo] = PACIFIC_TIME)
     the `timezone` argument to set the timezone to use if none was specified in
     the parsed string.
     """
+    # Handle dumb typos that might be in the dates on the page :(
+    if date_string.endswith('/202'):
+        date_string += '0'
+
```

Comment:
While we're here, you could check that date_string is in the form `"mm/dd/yyyy"`, or check against a set of date patterns the news sources use, and raise an error if it's not. Due to the typos in Contra Costa's source, I think we should leave `ARTICLE_TITLE_PATTERN` as it is and do a stricter check here.
```suggestion
    # Handle dumb typos that might be in the dates on the page :(
    DATE_PATTERN = re.compile(r'^\d{2}/\d{2}/\d{4}$')     # The date in mm/dd/yyyy format. 
    if not DATE_PATTERN.match(date_string):
        # start a list of if's to handle every typo we might find, starting with the '/202' typo
        if date_string.endswith('/202'):
            date_string += '0'
        else:
            raise FormatError("Date string does not match pattern 'dd/mm/yyyy' ")
```

Reply:
Hmmmm, if we did a check like this, it would have to move out of this function and directly into the Contra Costa scraper (this function handles a wide variety of formats; I thought this check was safe because I can’t think of any format that we would expect to end with “/202” or “/2020” besides mm/dd/yyyy).

An alternative might be to keep the check here, but not raise an error:

```suggestion
    DATE_PATTERN = re.compile(r'^\d+/\d+/\d+$')
    if DATE_PATTERN.match(date_string):
        if date_string.endswith('/202'):
            date_string += '0'
```

Or to check and fix the year *after* parsing (would work for this case, not sure what other odd future typos would be fixable this way):

```py
def parse_datetime(date_string: str, timezone: Optional[tzinfo] = PACIFIC_TIME) -> datetime:
    date = dateutil.parser.parse(date_string)

    # Handle a possible typo in the date string.
    if date.year == 202:
        date = date.replace(year=2020)

    # Gut-check whether this date seems reasonable
    if abs((datetime.datetime.now(date.tzinfo) - date).days) > 5 * 365:
        raise ValueError(f'Unknown date format: "{date_string}"')

    if date.tzinfo is None:
        date = date.replace(tzinfo=timezone)

    return date
```

Reply:
Updated this to a mix of the above approaches. I’m going to go ahead and merge since you already approved here, @elaguerta, but please comment if you think it should be revised further in a new PR.